src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.Windows.cs

```diff
@@ -14,871 +13,607 @@
 namespace System.Net.Security
 {
     //
-    // The class maintains the state of the authentication process and the security context.
-    // It encapsulates security context and does the real work in authentication and
+    // The class does the real work in authentication and
     // user data encryption with NEGO SSPI package.
     //
     // This is part of the NegotiateStream PAL.
     //
-    internal class NegoState
+    internal partial class NegoState
     {
-        private const int ERROR_TRUST_FAILURE = 1790;   // Used to serialize protectionLevel or impersonationLevel mismatch error to the remote side.
-
-        private static readonly byte[] s_emptyMessage = new byte[0];
-        private static readonly AsyncCallback s_readCallback = new AsyncCallback(ReadCallback);
-        private static readonly AsyncCallback s_writeCallback = new AsyncCallback(WriteCallback);
-
-        private Stream _innerStream;
-        private bool _leaveStreamOpen;
+        internal IIdentity GetIdentity()
+        {
+            CheckThrow(true);
 
-        private Exception _exception;
+            IIdentity result = null;
+            string name = _context.IsServer ? _context.AssociatedName : _context.Spn;
+            string protocol = "NTLM";
```

Comment:
Another example of a postponement I don't understand.  The fix here is to change:
```C#
string protocol = "NTLM";
protocol = _content.ProtocolName;
```
to:
```C#
string protocol = _content.ProtocolName;
```
It'll take you longer to add the relevant summary details to the issue then it would take to fix this.